chain/sync.go

```diff
@@ -511,10 +511,13 @@ func (syncer *Syncer) ValidateBlock(ctx context.Context, b *types.FullBlock) err
 			return xerrors.Errorf("failed to get sector size for block miner: %w", err)
 		}
 
-		snum := types.BigDiv(mpow, types.NewInt(ssize))
+		mps, err := stmgr.GetMinerProvingSet(ctx, syncer.sm, baseTs, h.Miner)
```

Comment:
This isn't a super cheap call, and the math we had here previously is exactly the same.

Reply:
It is not in a world in which you don't assume storage power across sectors is a multiple of `sectorSize` (if you get my gist....).

Lotus' call, but this is more future proof